### PR TITLE
Implement instant search screen - 5

### PR DESCRIPTION
**Pull request by msasikanth**
https://app.clubhouse.io/simpledotorg/story/2029/implement-instant-search-screen

### CODE REVIEW

**Review by Honey14 — CHANGES_REQUESTED** [review]
Looks good overall. App is crashing when clicked on `Register New Patient`.
Reproduction scenario :
* Click on enter patient's name -> search -> click on register new patient -> crash

Not the expected behaviour :
* Scan BP passport -> Register New patient -> does nothing just closes the sheet -> click on register new patient in `InstantSearchScreen` -> crash 
(Expected behaviour: opening patient entry screen from BpPassportSheet)

File: app/src/test/java/org/simple/clinic/instantsearch/InstantSearchUiRendererTest.kt [hunk 1]
```diff
@@ -9,18 +9,47 @@ class InstantSearchUiRendererTest {
...
+    // when
+    uiRenderer.render(resultsLoadedModel)
+
+    // then
+    verify(ui).hideSearchProgress()
+    verifyNoMoreInteractions(ui)
+  }
+
+  @Test
+  fun `when there is no instant search progress state, then hide search progress`() {
```
[the remark this answers is not in the document]
Comment (msasikanth) [reply app/src/test/java/org/simple/clinic/instantsearch/InstantSearchUiRendererTest.kt:43]: This is not a duplicate of it the done state, here we are verifying the initial state when screen/model is first created.

**Comment by msasikanth** [comment]
> Looks good overall. App is crashing when clicked on `Register New Patient`.
> Reproduction scenario :
> 
> * Click on enter patient's name -> search -> click on register new patient -> crash
> 
> Not the expected behaviour :
> 
> * Scan BP passport -> Register New patient -> does nothing just closes the sheet -> click on register new patient in `InstantSearchScreen` -> crash
>   (Expected behaviour: opening patient entry screen from BpPassportSheet)

Yes, that crash is expected. We are tracking it [here](https://app.clubhouse.io/simpledotorg/story/2209/remove-assertion-error-and-deprecate-setonfocuschangelistener-in-dateofbirthedittext), the story also contains the links to why the crash is happening. 

The BP passport sheet results have been just merged. So I need to add support for it now.

Since this feature is under feature flag, these should be fine. While I am making the final PR, I will enable the feature.

**Comment by msasikanth** [comment]
Address this [issue ](https://app.clubhouse.io/simpledotorg/story/2209/remove-assertion-error-and-deprecate-setonfocuschangelistener-in-dateofbirthedittext) as well

**Review by Honey14 — CHANGES_REQUESTED** [review]
Looks good overall. Because of the latest date issue change, adding new blood sugar is not working as expected in the PR but is working in the master. PTAL.

**Comment by msasikanth** [comment]
> Looks good overall. Because of the latest date issue change, adding new blood sugar is not working as expected in the PR but is working in the master. PTAL.

Can you explain what you mean by not working as expected?

**Comment by Honey14** [comment]
> > Looks good overall. Because of the latest date issue change, adding new blood sugar is not working as expected in the PR but is working in the master. PTAL.
> 
> Can you explain what you mean by not working as expected?

Reproduction -> click on add diabetes -> add any rbs -> does not work at all. 
Reason : My bad it's not the date change. Logs don't say much. Can we check what was the last change we made?

**Review by Honey14 — APPROVED** [review]
Looks good

**Pull request merged**